zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/transform/ResourceTransformer.java

```diff
@@ -0,0 +1,195 @@
+              }
+              writeResourceRecord(metadata, resource);
+            });
+  }
+
+  private void writeResourceRecord(
+      final ResourceMetadataRecord resourceMetadataRecord, final DeploymentResource resource) {
+    stateWriter.appendFollowUpEvent(
+        resourceMetadataRecord.getResourceKey(),
+        ResourceIntent.CREATED,
+        new ResourceRecord().wrap(resourceMetadataRecord, resource.getResource()));
+  }
+
+  private void appendMetadataToResourceRecord(
+      final ResourceMetadataRecord resourceMetadataRecord,
+      final Resource resource,
+      final DeploymentResource deploymentResource,
+      final DeploymentRecord deploymentRecord) {
+    final LongSupplier newResourceKey = keyGenerator::nextKey;
+    final DirectBuffer checksum =
+        checksumGenerator.checksum(deploymentResource.getResourceBuffer());
+    final String tenantId = deploymentRecord.getTenantId();
+
+    resourceMetadataRecord.setResourceId(resource.id);
+    resourceMetadataRecord.setChecksum(checksum);
+    resourceMetadataRecord.setResourceName(deploymentResource.getResourceName());
+    resourceMetadataRecord.setTenantId(tenantId);
+    Optional.ofNullable(resource.versionTag).ifPresent(resourceMetadataRecord::setVersionTag);
+
+    resourceState
+        .findLatestResourceById(resourceMetadataRecord.getResourceId(), tenantId)
+        .ifPresentOrElse(
+            latestResource -> {
+              final boolean isDuplicate =
+                  latestResource.getChecksum().equals(resourceMetadataRecord.getChecksumBuffer())
+                      && latestResource
+                          .getResourceName()
+                          .equals(resourceMetadataRecord.getResourceNameBuffer());
+
+              if (isDuplicate) {
+                final int latestVersion = latestResource.getVersion();
+                resourceMetadataRecord
+                    .setResourceKey(latestResource.getResourceKey())
+                    .setVersion(latestVersion)
+                    .setDeploymentKey(latestResource.getDeploymentKey())
+                    .setDuplicate(true);
+              } else {
+                resourceMetadataRecord
+                    .setResourceKey(newResourceKey.getAsLong())
+                    .setVersion(resourceState.getNextResourceVersion(resource.id, tenantId))
+                    .setDeploymentKey(deploymentRecord.getDeploymentKey());
+              }
+            },
+            () ->
+                resourceMetadataRecord
+                    .setResourceKey(newResourceKey.getAsLong())
+                    .setVersion(INITIAL_VERSION)
+                    .setDeploymentKey(deploymentRecord.getDeploymentKey()));
+  }
+
+  private Either<Failure, Resource> parseResource(final DeploymentResource resource) {
+    try {
+      final var res = JSON_MAPPER.readValue(resource.getResource(), Resource.class);
+      return validateResource(res);
```

Comment:
❌  This code assumes that a resource file has JSON content. That's only expected if the resource file is a robot (RPA) script, but it doesn't have to be. And the whole `ResourceTransformer` class will be applied to any resources that are not BPMN, DMN, Form, or XML according to the usage of [this method](https://github.com/camunda/camunda/pull/25918/files#diff-b647b3d6d8c4f102119b11dfcaec9673c86cd3955efe6226c121495e12e070f0R223-R226).

I think we should re-name this transformer into a `RobotScriptTransformer` or `RpaTransformer`. Persisting into the `ResourceState` still makes sense, but the transformation logic should remain specific for RPA scripts.